### PR TITLE
Message match

### PR DESCRIPTION
**Pull request by mrsylerpowers**

### CODE REVIEW

File: packages/server/src/server/api/v1/apple/actions.ts [hunk 1]
```diff
@@ -166,7 +166,7 @@ export class ActionHandler {
         // Create the awaiter
         let messageAwaiter = null;
         if (isNotEmpty(message)) {
-            messageAwaiter = new MessagePromise(chatGuid, message, false, now, tempGuid);
+            messageAwaiter = new MessagePromise({chatGuid: chatGuid, text:message, isAttachment: false, sentAt:now,  tempGuid:tempGuid});
```
Comment (zlshames) [inline packages/server/src/server/api/v1/apple/actions.ts:169]: please add a space after all colons here

File: packages/server/src/server/api/v1/interfaces/messageInterface.ts [hunk 2]
```diff
@@ -204,7 +204,7 @@ export class MessageInterface {
 
         // We need offsets here due to iMessage's save times being a bit off for some reason
         const now = new Date(new Date().getTime() - 10000).getTime(); // With 10 second offset
-        const awaiter = new MessagePromise(chatGuid, messageText, false, now, tempGuid);
+        const awaiter = new MessagePromise({chatGuid: chatGuid, text:messageText, isAttachment:false, sentAt:now, tempGuid:tempGuid});
```
Comment (zlshames) [inline packages/server/src/server/api/v1/interfaces/messageInterface.ts:207]: spaces

**Review by zlshames — CHANGES_REQUESTED** [review]

File: packages/server/src/server/managers/outgoingMessageManager/messagePromise.ts [hunk 3]
```diff
@@ -117,3 +117,11 @@ export class MessagePromise {
         return this.text === onlyAlphaNumeric(message.text) && this.sentAt < message.dateCreated.getTime();
     }
 }
+interface MessagePromiseConstructorParameters {
+    chatGuid: string;
+    text: string;
+    isAttachment: boolean;
+    sentAt: Date | number;
```
[the remark this answers is not in the document]
Comment (mrsylerpowers) [reply packages/server/src/server/managers/outgoingMessageManager/messagePromise.ts:124]: L51 is where this is done

[the remark this answers is not in the document]
Comment (mrsylerpowers) [reply packages/server/src/server/managers/outgoingMessageManager/messagePromise.ts:124]: Always passed as a number I think

[the remark this answers is not in the document]
Comment (zlshames) [reply packages/server/src/server/managers/outgoingMessageManager/messagePromise.ts:124]: If it's always a number, and we don’t manually handle dates, can we just make the type be a number?

[the remark this answers is not in the document]
Comment (mrsylerpowers) [reply packages/server/src/server/managers/outgoingMessageManager/messagePromise.ts:124]: Yes. I definitely think that is a better solution

**Pull request merged**